src/link.rs

```diff
@@ -62,13 +62,41 @@ fn collect_roots(
     roots
 }
 
+/// Ensure that all `CrateIndex`es have consistent schema versions. If all
+/// indexes have the same schema version, return the version number. Otherwise,
+/// return an error.
+///
+/// Precondition: there must be at least one `CrateIndex`. If this precondition
+/// is not met, this function will panic.
+fn check_equal_versions(
+    indexes: &[CrateIndex],
+) -> serde_cbor::Result<u64> {
+    match indexes.split_first() {
+        Some((first_index, last_indexes)) => {
+            let version = first_index.version;
+            for index in last_indexes {
+                if version != index.version {
+                    return Err(serde::ser::Error::custom(format!(
+                        "Mismatched schema version numbers ({:?} and {:?})",
+                        version,
+                        index.version,
+                    )));
+                }
+            }
+            Ok(version)
+        },
+        None =>
+            panic!("check_equal_versions: Empty list of crate indexes"),
```

Comment:
I think it would be legal to just return `SCHEMA_VER` here.  Then linking together zero inputs produces a valid empty file matching the current schema.

Reply:
This `None` case no longer exists in the new version of the code.